src/main/webapp/app/exercises/shared/result/updating-result.component.ts

```diff
@@ -123,8 +134,26 @@ export class UpdatingResultComponent implements OnChanges, OnDestroy {
                 ),
                 tap(({ submissionState }) => {
                     this.isBuilding = submissionState === ProgrammingSubmissionState.IS_BUILDING_PENDING_SUBMISSION;
+
+                    if (submissionState === ProgrammingSubmissionState.HAS_FAILED_SUBMISSION) {
+                        this.missingResultInfo = this.generateMissingResultInfoForFailedProgrammingExerciseSubmission();
+                    } else {
+                        // everything ok, remove the warning
+                        this.missingResultInfo = undefined;
+                    }
                 }),
             )
             .subscribe();
     }
+
+    private generateMissingResultInfoForFailedProgrammingExerciseSubmission() {
```

Comment:
I'd like to question this implementation. The `message` attribute is static and the second attribute only differs in two cases.

=> I'd suggest creating an enum containing the following cases: result available, result not available - offline solution, result not available - online solution.
This would mean that you move the whole texts to the component where they actually get used.
I think this will also clear up some other things.

What do you think?

Reply:
Yes, requires changing the enum and component for every case, but I guess this will not happen often (or knowing Artemis anytime in near future).